disnake/state.py

```diff
@@ -797,6 +805,13 @@ def parse_message_delete_bulk(self, data: gateway.MessageDeleteBulkEvent) -> Non
             found_messages = []
         raw.cached_messages = found_messages
         self.dispatch("raw_bulk_message_delete", raw)
+        guild = self._get_guild(raw.guild_id)
+        thread = guild and guild.get_thread(raw.channel_id)
+        for _ in found_messages:
+            if thread:
+                thread.message_count -= 1
+            else:
+                continue
```

Comment:
They don't all have to be cached, and bulk deletes all occur in the same channel. As such, we can subtract `len(data['ids'])` from the thread without having a for-loop.

```suggestion
        guild = self._get_guild(raw.guild_id)
        thread = guild and guild.get_thread(raw.channel_id)
        if thread:
            thread.message_count -= len(raw.message_ids)
```